crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -63,18 +69,48 @@ impl TimelineBuilder {
     /// Enable tracking of the fully-read marker and the read receipts on the
     /// timeline.
     pub(crate) fn track_read_marker_and_receipts(mut self) -> Self {
-        self.track_read_marker_and_receipts = true;
+        self.settings.track_read_receipts = true;
+        self
+    }
+
+    /// Use the given filter to choose whether to add events to the timeline.
+    ///
+    /// # Arguments
+    ///
+    /// * `filter` - A function that takes a deserialized event, and should
+    ///   return `true` if the event should be added to the `Timeline`.
+    ///
+    /// If this is not overridden, the timeline uses the default filter that
+    /// allows every event.
+    ///
+    /// Note that currently not all event types have a representation as a
+    /// `TimelineItem` so these are not added no matter what the filter returns.
+    /// It is also not possible to filter out `m.room.encrypted` events
+    /// otherwise they won't be decrypted when the appropriate room key arrives.
```

Comment:
In my mind, the "currently" from the previous sentence carried on to this one. It was documented as much for us as for users

Reply:
How about this then:

```md
Note that currently

- not all event types [...]
- it is not possible to filter out `m.room.encrypted` events (otherwise they couldn't by decrypted when the appropriate room key arrives)
```